src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,104 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Movie {
+  title: string;
+  description: string;
+  imgUrl: string;
+  imdbUrl: string;
+  imdbId: string;
+}
+
+interface NewMovieProps {
+  onAddMovie: (newMovie: Movie) => void;
+}
+
+export const NewMovie = ({ onAddMovie }: NewMovieProps) => {
+  const [count, setCount] = useState(0);
+
+  const [formData, setFormData] = useState<Movie>({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const handleChange = (fieldName: keyof Movie, newValue: string) => {
+    setFormData(prevState => ({ ...prevState, [fieldName]: newValue }));
+  };
+
+  const isFormValid = () => {
+    const { title, imgUrl, imdbUrl, imdbId } = formData;
+
+    return title && imgUrl && imdbUrl && imdbId;
+  };
```

Comment:
you can write it just like a variable 

And with this case, you can use a property in your jsx


```suggestion

    const { title, imgUrl, imdbUrl, imdbId } = formData;
 
    const isFormValid = title && imgUrl && imdbUrl && imdbId;
```